tests/snuba/api/endpoints/test_organization_events_stats_mep.py

```diff
@@ -1280,6 +1280,112 @@ def test_apdex_issue(self):
             [{"count": 0.5}],
         ]
 
+    def test_glob_http_referer_on_demand(self):
+        agg = "count()"
+        network_id_tag = "networkId"
+        url = "https://sentry.io"
+        query = f'http.url:{url}/*/foo/bar/* http.referer:"{url}/*/bar/*" event.type:transaction'
+        spec = OnDemandMetricSpec(
+            field=agg,
+            groupbys=[network_id_tag],
+            query=query,
+            spec_type=MetricSpecType.DYNAMIC_QUERY,
+        )
+        assert spec.to_metric_spec(self.project) == {
+            "category": "transaction",
+            "mri": "c:transactions/on_demand@none",
+            "field": None,
+            "tags": [
+                {"key": "query_hash", "value": "fc0c932e"},
+                {"key": "networkId", "field": "event.tags.networkId"},
+                {"key": "environment", "field": "event.environment"},
+            ],
+            "condition": {
+                "op": "and",
+                "inner": [
+                    {
+                        "op": "glob",
+                        "name": "event.request.url",
+                        "value": ["https://sentry.io/*/foo/bar/*"],
+                    },
+                    {
+                        "op": "glob",
+                        "name": "event.request.referer",
+                        "value": ["https://sentry.io/*/bar/*"],
+                    },
+                ],
+            },
+        }
+
+        for hour in range(0, 5):
+            self.store_on_demand_metric(
+                1,
+                spec=spec,
+                additional_tags={network_id_tag: "1234"},
+                timestamp=self.day_ago + timedelta(hours=hour),
+            )
+            self.store_on_demand_metric(
+                1,
+                spec=spec,
+                additional_tags={network_id_tag: "5678"},
+                timestamp=self.day_ago + timedelta(hours=hour),
+            )
+
+        response = self.do_request(
```

Comment:
In the original customer case, the UI calls with `statsPeriod:24hr` which has a bug. 
Filed here: https://github.com/orgs/getsentry/projects/156/views/2?pane=issue&itemId=46553631